### PR TITLE
[:pineapple: < T0908-MG] Hardcode example queries by using batching / multiframe

### PR DESCRIPTION
**Pull request by 42jeremy**
[project-pineapples < T0908-MG] PR
What was done?

Implemented operators working with several frames at once (batching):
- OnceOperator
- ScanAllOperator
- ScanAllByLabelOperator
- ScanAllByLabelPropertyValueOperator
- ScanAllByIdOperator
- ProduceOperator

Architecture of ExpandOperator is ready but implementation is not working  yet.

### CODE REVIEW

File: src/query/v2/plan/operator.cpp [hunk 1]
```diff
@@ -78,12 +78,16 @@ extern const Event OnceOperator;
 extern const Event CreateNodeOperator;
 extern const Event CreateExpandOperator;
 extern const Event ScanAllOperator;
+extern const Event ScanAllOperator_Distributed;
```
Comment (gitbuda) [inline src/query/v2/plan/operator.cpp:81]: Quick question, why with `_` (`_Distributed`)?

Comment (42jeremy) [reply src/query/v2/plan/operator.cpp:81]: It's temporary, the PR is still as a draft for now

File: src/query/v2/db_accessor.hpp [hunk 2]
```diff
@@ -209,6 +209,7 @@ inline bool EdgeAccessor::IsCycle() const { return To() == From(); }
 class DbAccessor final {
   storage::v3::Storage::Accessor *accessor_;
 
+ public:  // #NoCommit temporary to continue working
```
Comment (42jeremy) [inline src/query/v2/db_accessor.hpp:212]: Needs to be removed

**Comment by 42jeremy** [comment]
All  #NoCommit need to be removed

File: src/query/v2/plan/operator_distributed.cpp [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+    : ScanAll(input, output_symbol, view), expression_(expression) {
+  MG_ASSERT(expression);
+}
+
+ACCEPT_WITH_INPUT(ScanAllById)
+
+UniqueCursorPtr ScanAllById::MakeCursor(utils::MemoryResource *mem) const {
+  EventCounter::IncrementCounter(EventCounter::ScanAllByIdOperator);
+
+  auto vertices = [this](Frames &frames, ExecutionContext &context) -> std::optional<std::vector<VertexAccessor>> {
```
Comment (42jeremy) [inline src/query/v2/plan/operator_distributed.cpp:329]: Double check implementation.
If not correct, just remove as we don't need it right now.

**Comment by 42jeremy** [comment]
@gitbuda @kostasrim if you wish to start the review, do not focus yet on the expand part of it. It is not working as expected yet.

File: src/query/v2/plan/operator_distributed.lcp [hunk 4]
```diff
@@ -0,0 +1,800 @@
...
+     std::vector<std::pair<uint64_t, std::shared_ptr<LogicalOperator>>>
+         loaded_ops;
+   };
+
+   struct SlkLoadHelper {
+     AstStorage ast_storage;
+     std::vector<std::shared_ptr<LogicalOperator>> loaded_ops;
+   };
+
+   virtual void PerformFullEnumeration() {} // #NoCommit =0 by default, only temporary so peers can work: all operators should implement it, perhaps it's not only about ScanAll
```
Comment (42jeremy) [inline src/query/v2/plan/operator_distributed.lcp:192]: I am pretty sure the implement should be like this by default:
```
virtual void PerformFullEnumeration() { 
if (HasSingleInput()) {
   input()->PerformFullEnumeration();
   }
}
```

**Pull request closed without merging**